scipy/stats/tests/test_distributions.py

```diff
@@ -3293,7 +3266,12 @@ def test_signature_inspection_2args(self):
 
     def test_signature_inspection_2args_incorrect_shapes(self):
         # both _pdf and _cdf defined, but shapes are inconsistent: raises
-        assert_raises(TypeError, _distr3_gen, name='dummy')
+        try:
+            _distr3_gen(name='dummy')
+        except TypeError:
+            pass
+        else:
+            raise AssertionError('TypeError not raised.')
```

Comment:
I don't understand this one--we've gone from a standard ecosystem `assert_raises` to a hand-coded block that does something similar?

Reply:
Btw I didn't make this change and I'm not sure how it got introduced into this PR. Perhaps at some point when I pulled in master it picked these up?

Reply:
This looks like leftover code, which was cleaned up on master, and crept back pin s rebase or some such.

Reply:
Spurious non-PR code now removed.